src/configure.py

```diff
@@ -760,14 +769,50 @@ def load_config( config ):
             try:
                paths[temp[0]] = temp[1]
             except:
-               paths[temp[0]] = ''
+               paths[temp[0]] = ""
 
     return paths, compilers, flags, gpus
 
-def set_conditional_defaults( args ):
-    if args['machine'] is None:
-        args['machine'] = get_machine()
+def load_setting():
+    """
+    Return the default machine setting in the following order:
+    1. Read the local  setting located at `GAMER_LOCAL_SETTING`.
+    2. Read the global setting located at `GAMER_GLOBAL_SETTING`.
+    3. Fall back to `eureka_intel` for backward compatibility.
+
+    Format of the setting file:
+    1. Comment starts with `#`.
+    2. Separate the variable name and value with space. The line starts with the variable name.
+    3. Only the fisrt value of the line will be loaded.
+    4. Only the last variable of duplicated variables will be loaded.
+    """
+    sys_setting  = SystemSetting()
+    setting_file = None
 
+    if os.path.isfile( GAMER_GLOBAL_SETTING ): setting_file = GAMER_GLOBAL_SETTING
+    if os.path.isfile( GAMER_LOCAL_SETTING  ): setting_file = GAMER_LOCAL_SETTING
+
+    if setting_file is None:
+        LOGGER.info("System setting file not detected.")
+        return sys_setting
+    else:
+        LOGGER.info("Using %s as setting file."%(setting_file))
```

Comment:
```suggestion
    if   os.path.isfile( GAMER_LOCAL_SETTING  ): setting_file = GAMER_LOCAL_SETTING
    elif os.path.isfile( GAMER_GLOBAL_SETTING ): setting_file = GAMER_GLOBAL_SETTING
    else:
        LOGGER.info("System setting file not detected.")
        return sys_setting

    LOGGER.info("Using %s as setting file."%(setting_file))
```
Not necessary, but make the fall back logic cleaner.

Reply:
In a more pythonic style, we can remove `setting_file = None` at the beginning.
But this violate the variable life cycle rule in `c/c++`.
Should we do this file in a pythonic flavor or a c/c++ flavor?

Reply:
Agree. Let's remove `setting_file = None`.